rewrite-hcl/src/main/java/org/openrewrite/hcl/tree/Hcl.java

```diff
@@ -1531,20 +1533,51 @@ public enum Type {
 
     @FieldDefaults(makeFinal = true, level = AccessLevel.PRIVATE)
     @EqualsAndHashCode(callSuper = false, onlyExplicitlyIncluded = true)
-    @Data
+    @RequiredArgsConstructor
+    @AllArgsConstructor(access = AccessLevel.PRIVATE)
     class TemplateInterpolation implements Hcl, Expression {
+        @Nullable
+        @NonFinal
+        transient WeakReference<Padding> padding;
+
         @With
         @EqualsAndHashCode.Include
+        @Getter
         UUID id;
 
         @With
+        @Getter
         Space prefix;
 
         @With
+        @Getter
         Markers markers;
 
-        @With
-        Expression expression;
+        HclRightPadded<Expression> expression;
+
+        @JsonCreator
+        TemplateInterpolation(@JsonProperty("id") UUID id,
+                              @JsonProperty("prefix") Space prefix,
+                              @JsonProperty("markers") Markers markers,
+                              @JsonProperty("expression") Object expression) {
+            this.id = id;
+            this.prefix = prefix;
+            this.markers = markers;
+            if (expression instanceof Expression) {
+                this.expression = HclRightPadded.withElement(null, (Expression) expression);
+            } else {
+                //noinspection unchecked
+                this.expression = (HclRightPadded<Expression>) expression;
+            }
+        }
```

Comment:
@knutwannheden hope I got this right based on your comment above in https://github.com/openrewrite/rewrite/pull/6076/files#r2391939772